### ISSUE TITLE
Pytest: yield tests are deprecated and must be removed.

### ISSUE THREAD

**Issue by sciunto** [scikit-image/scikit-image#2627]
## Description

Cc @NelleV 

```
WC1 /home/travis/build/sciunto/scikit-image/skimage/exposure/tests/test_exposure.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/exposure/tests/test_exposure.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/exposure/tests/test_exposure.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/exposure/tests/test_exposure.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/exposure/tests/test_exposure.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/exposure/tests/test_exposure.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/external/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/feature/tests/test_template.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/feature/tests/test_template.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/filters/tests/test_edges.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/filters/tests/test_edges.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/filters/tests/test_edges.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/filters/tests/test_edges.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/filters/tests/test_edges.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/filters/tests/test_edges.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/graph/tests/test_mcp.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/io/tests/test_tifffile.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_binary.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_binary.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_binary.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_binary.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_binary.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_binary.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_grey.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_grey.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_grey.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_grey.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_grey.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_grey.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_grey.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_grey.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_skeletonize_3d.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_skeletonize_3d.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_skeletonize_3d.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_skeletonize_3d.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_skeletonize_3d.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/morphology/tests/test_skeletonize_3d.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/restoration/tests/test_unwrap.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/restoration/tests/test_unwrap.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/restoration/tests/test_unwrap.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/restoration/tests/test_unwrap.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/restoration/tests/test_unwrap.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/restoration/tests/test_unwrap.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/restoration/tests/test_unwrap.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/restoration/tests/test_unwrap.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/restoration/tests/test_unwrap.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/transform/tests/test_radon_transform.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
WC1 /home/travis/build/sciunto/scikit-image/skimage/util/tests/test_dtype.py yield tests are deprecated, and scheduled to be removed in pytest 4.0
```

**Comment by NelleV** [comment]
Yep, I saw. It's very annoying :(

**Comment by nelzas** [comment]
Hello,

I'm volunteering to work on this, if nobody started yet.
I've currently removed the yields in one file, test_exposure.py, and used instead pytest paremtrize decorator.
See https://github.com/nelzas/scikit-image/commit/592edf5e63723c8110b96f7ebff56688f2ccff40
I'll be happy to get feedback whether to continue and if yes, improvement suggestions.
For example, maybe we can skip using a temp variable out:
instead of:
```
out = intensity_range(image, range_values=test_input)
assert_array_equal(out, expected)
```

one can do:
`assert_array_equal(intensity_range(image, range_values=test_input), expected)`

**Comment by soupault** [comment]
I'd like to see https://github.com/scikit-image/scikit-image/pull/2278 merged first.

>For example, maybe we can skip using a temp variable out

Rather not (IMO) :smile: .

**Comment by nelzas** [comment]
I continued to work on this before seeing your answer.
I did not finish but got rid of more than 100 warnings. It's in my fork.
I understand there are priorities, so even if I finish all 'yields' I won't open a pull request until I get a green light :-)

**Comment by sciunto** [comment]
@soupault I don't understand why you want to see #2278 merged first.

**Comment by nelzas** [comment]
Update: I've removed all yields from tests, this removed all 175 warnings of this kind.
In the majority of tests, I've used pytest parametrize decorator for this. But in three tests (test_template, test_unwrap_2d, test_unwrap_3d) I didn't use parametrize, and just replaced the yield with a call to the relevant check function, so as a result the count of tests went down by 3.
As I've said before I won't open PR before getting a green light.

**Comment by soupault** [comment]
@sciunto LOL, sorry, I meant https://github.com/scikit-image/scikit-image/pull/2614 :).

**Comment by stefanv** [comment]
@nelzas Thank you for investigating this issue!  Please go ahead and open a PR; that'd be much appreciated.

**Comment by nelzas** [comment]
OK, I've opened the PR. I will be glad to receive comments if things need to be done in a better way.

**Comment by soupault** [comment]
Closed via #2824
